GAE/src/org/waterforpeople/mapping/dataexport/RawDataSpreadsheetImporter.java

```diff
@@ -154,44 +208,97 @@ public void runImport(File file, String serverBase, Map<String, String> criteria
      *
      * @param sheet
      * @param startRow
+     * @param columnIndexToQuestionId
+     * @param questionIdToQuestionDto
      * @return InstanceData
      */
-    public InstanceData parse(Sheet sheet, int startRow) {
+    public InstanceData parseInstance(Sheet sheet, int startRow,
+            Map<Long, QuestionDto> questionIdToQuestionDto, Map<Long, Long> columnIndexToQuestionId) {
 
         // File layout
         // 0. SurveyedLocaleIdentifier
+        // 7. Repeat
         // 1. SurveyedLocaleDisplayName
         // 2. DeviceIdentifier
         // 3. SurveyInstanceId
         // 4. CollectionDate
         // 5. SubmitterName
         // 6. SurveyalTime
-        // 7. Repeat
+
         // 8 - N. Questions
         // N + 1. Digest
 
         Row baseRow = sheet.getRow(startRow);
-        String surveyedLocaleIdentifier = baseRow.getCell(0).getStringCellValue();
-        String surveyedLocaleDisplayName = baseRow.getCell(1).getStringCellValue();
-        String deviceIdentifier = baseRow.getCell(2).getStringCellValue();
-        String surveyInstanceId = baseRow.getCell(3).getStringCellValue();
-        String collectionDate = baseRow.getCell(4).getStringCellValue();
-        String submitterName = baseRow.getCell(5).getStringCellValue();
-        String surveyalTime = baseRow.getCell(6).getStringCellValue();
-        int firstQuestionColumn = 8;
+        if (baseRow == null) {
+            return null;
+        }
+        String surveyedLocaleIdentifier = ExportUtils.parseCellAsString(baseRow.getCell(0));
+        String surveyedLocaleDisplayName = ExportUtils.parseCellAsString(baseRow.getCell(2));
+        String deviceIdentifier = ExportUtils.parseCellAsString(baseRow.getCell(3));
+        String surveyInstanceId = ExportUtils.parseCellAsString(baseRow.getCell(4));
+        Date collectionDate = ExportUtils.parseDate(
+                ExportUtils.parseCellAsString(baseRow.getCell(5)));
+        String submitterName = ExportUtils.parseCellAsString(baseRow.getCell(6));
+        String surveyalTime = ExportUtils.parseCellAsString(baseRow.getCell(7));
 
         int iterations = 1;
 
         // Count the number maximum number of iterations for this instance
         while (true) {
             Row row = sheet.getRow(startRow + iterations);
-            if (row == null || row.getCell(7).getStringCellValue().equals("1")) {
+            if (row == null || row.getCell(1).getNumericCellValue() == 1.0) {
```

Comment:
In case we have an empty row like what appears in the reports with blank rows where the iterations were deleted, then `row == null` means the next survey instance will start processing but on a row that actually contains data for another iteration of repeat group responses, right?